### DISCUSSION TITLE
Bugs in preview comments (on beta.nextjs.org)

### DISCUSSION THREAD

**Discussion by mack-useo** [vercel/community#1666] [Help]
### Summary

I'm researching new features in Next.js on the beta docs , and I'm trying to be active in the comments, which use Vercel preview comments (as far as I can tell). 

So far, it's been very frustrating to use comments on beta.nextjs.org because: 
- comments get detached from the page (only accessible via inbox), so we no longer know to what they referred - a quote of the selected text (or even the entire paragraph) would greatly improve the usability
  - it happens to resolved comments (which otherwise could've been a valuable source of information)
  - it also happens to some unresolved comments, presumably after the content is changed in a new deployment (but it might happen if a paragraph is moved, so the comment could still be valid)
- pretty quickly, the whole commenting functionality starts lagging (for example marking the comment as read takes a few minutes after opening it)
  - it might be connected to the memory footprint - as soon as I start using the inbox, the tab jumps from using ~150MB to 1-2GB of memory (according to the Chrome task manager)
- I often cannot use the comments at all - when that happens, I'm getting this error on refreshing the page or logging in: `{"error":{"message":"Deadlock detected","code":"WRAPPED_DEADLOCK"}}`
- the inbox has a dropdown that shows comments from all pages, but when I click the link box thing (the URL preview with the title of the page where the comment was left), the page freezes for about 10 seconds and then switches to that linked page, but the commenting UI disappears, and I get more errors in the console (they include [this link](https://reactjs.org/docs/error-decoder.html/?invariant=185))
- scrolling the comment into view on mouse hover of a comment in the inbox is a terrible experience, as it keeps scrolling away when I accidentally move my cursor over some comment
- it's very easy to resolve a comment by mistake, as it takes just one click - that's less than is needed for following a thread (which should be one click)

I know it might be a unique case, with the number of comments there, but at this stage, it's just frustrating to try and keep active in the comments. 

I confirmed all of these problems after updating Chrome and in an incognito window (without extensions), and the issues persist. 

Is there any work being done on these issues? 
Can I do something to make them work better (like using a different browser or something)? 

System info: 
- Chrome 110.0.5481.178 (both regular and incognito)
- Windows 10 Pro (10.0.19044 Build 19044)
- screenshot extension: never installed

### Example

https://beta.nextjs.org/docs/rendering/server-and-client-components#thread-id=aBpsX

**Comment by amyegan** [comment]
Thanks for providing this detailed feedback. I shared it with the Preview Comments team 🙇

**Reply by mack-useo** [reply]
Thank you, @amyegan! 

Feel free to let me know if you need more details or if anything on my side might be causing these issues.

**Reply by mack-useo** [reply]
@amyegan - is there any news on the topic? I don't expect it to be fixed already, but have you guys been able to replicate this or find a workaround (i.e. some browsers handle it better or something)?

**Reply by amyegan** [reply]
I don't have any specific updates to share yet, but the Preview Comments team is working on performance improvements to address this problem. Thanks for checking!
